src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -1808,84 +1798,91 @@ def get_state(self):
 
     def __init__(
         self,
         service: DatamodelService,
         rules: str,
-        query: str,
-        path: Path = None,
-    ):
-        """__init__ method of PyQuery class."""
+        name: str,
+        path: Path | None = None,
+    ) -> None:
+        """__init__ method of _PyAction class."""
         self.service = service
         self.rules = rules
-        self.query = query
-        if path is None:
-            self.path = []
-        else:
-            self.path = path
+        setattr(self, self._operation, name)
+        self.path = [] if path is None else path
 
     def __call__(self, *args, **kwds) -> Any:
-        """Execute the query.
-
-        Returns
-        -------
-        Any
-            Return value.
-        """
-        return self.service.execute_query(
-            self.rules, convert_path_to_se_path(self.path), self.query, kwds
+        """Execute the operation (command or query)."""
+        execute_method = getattr(self.service, f"execute_{self._operation}")
+        return execute_method(
+            self.rules,
+            convert_path_to_se_path(self.path),
+            getattr(self, self._operation),
+            kwds,
         )
 
-    def help(self) -> None:
-        """Prints help string."""
-        response = self.service.get_specs(
-            self.rules, convert_path_to_se_path(self.path)
-        )
-        help_string = _get_value_from_message_dict(
-            response, [member_specs_oneof_fields, "query", "helpstring"]
+    def _create_arguments(self) -> str:
+        """Create arguments for this operation."""
+        return self.service.create_command_arguments(
+            self.rules,
+            convert_path_to_se_path(self.path),
+            getattr(self, self._operation),
         )
-        print(help_string)
+
+    def _get_create_instance_args(self):
+        """Prepare arguments for PyArguments constructor."""
+        try:
+            id = self._create_arguments()
+            return [
+                self.service,
+                self.rules,
+                getattr(self, self._operation),
+                self.path.copy(),
+                id,
+            ]
+        except (RuntimeError, ValueError) as e:
+            logger.warning(
+                f"datamodels_se.{self.__class__.__name__} "
+                f"was unable to construct {self._operation} arguments. "
+                "This may be due to gRPC issues or unsupported Fluent version "
+                "(23.1+ required). Error details: %s",
+                e,
+            )
+
+    def create_instance(self) -> "PyArguments":
+        """Create an operation instance."""
+        args = self._get_create_instance_args()
+        if args is not None:
+            return PyArguments(*args)
+
+
+class PyQuery(_PyAction):
+    """Query class using the StateEngine-based DatamodelService as the backend. Use this
+    class instead of directly calling the DatamodelService's method."""
+
```

Comment:
I notice that we have other similar docstrings elsewhere in this module. The docstring is too heavily into unimportant implementation details.

The style should be more user-centric; e.g., "Enables querying Fluent’s data model through a simple Python interface."